src/components/context/utilities.ts

```diff
@@ -1,60 +1,73 @@
-import { List } from 'immutable';
 import { v4 as uuidv4 } from 'uuid';
 
-import { Member, convertJs } from '@graasp/sdk';
-import { AppDataRecord, AppSettingRecord } from '@graasp/sdk/frontend';
+import { AppData, AppSetting, Member } from '@graasp/sdk';
 
 import {
+  APP_SETTING_NAMES,
+  AppSettingName,
   DEFAULT_APP_DATA_VALUES,
   FAILURE_MESSAGES,
   QuestionType,
 } from '../../config/constants';
 import {
-  AppDataDataRecord,
-  AppDataQuestionRecord,
-  FillTheBlanksAppDataDataRecord,
-  MultipleChoiceAppDataDataRecord,
-  QuestionDataAppSettingRecord,
-  QuestionDataRecord,
-  SliderAppDataDataRecord,
-  TextAppDataDataRecord,
+  AppDataQuestion,
+  FillTheBlanksAppDataData,
+  MultipleChoiceAppDataData,
+  QuestionAppDataData,
+  QuestionData,
+  QuestionDataAppSetting,
+  QuestionListType,
+  SliderAppDataData,
+  TextAppDataData,
 } from '../types/types';
 
 export const generateId = (): string => {
   return uuidv4();
 };
 
-export const getQuestionById = (
-  data: List<QuestionDataAppSettingRecord>,
-  id: string
-) => {
+export const getQuestionById = (data: QuestionDataAppSetting[], id: string) => {
   return data.find((d) => d.data.questionId === id);
 };
```

Comment:
It would be nice to give this function a generic type parameter that would let us case the data type. To centralize the casting for example.